aee.js

```diff
@@ -23,3 +23,15 @@ Array.prototype.any = function (spec) {
   }
   return false
 }
+
+Array.prototype.take = function (howMany, spec) {
+  let result = []
+  if (howMany <= 0) return result
+  const len = this.length
+  const searchSpec = (typeof spec === 'function') ? spec : () => true
```

Comment:
yeah, and, you might keep the `let result =[]` on the top. weird but it can actually be a `const` too.